src/main/scala/com/mwz/sonar/scala/ScalaPlugin.scala

```diff
@@ -46,16 +48,42 @@ object Scala {
   private val FileSuffixesPropertyKey = "sonar.scala.file.suffixes"
   private val DefaultFileSuffixes = Array(".scala")
   private val ScalaVersionPropertyKey = "sonar.scala.version"
-  private val DefaultScalaVersion = ScalaVersions.Scala_2_11.toString()
+  private val DefaultScalaVersion = ScalaVersions.Scala_2_11
+  private val ScalaVersionPattern = """(\d+)\.(\d+)(?:\..+)?""".r
   private val SourcesPropertyKey = "sonar.sources"
   private val DefaultSourcesFolder = "src/main/scala"
 
-  def getScalaVersion(settings: Configuration): String =
-    settings.get(ScalaVersionPropertyKey).toOption.getOrElse(DefaultScalaVersion)
+  private val logger = Loggers.get(classOf[Scala])
+
+  def getScalaVersion(settings: Configuration): ScalaVersion = {
+    def parseVersion(s: String): Option[ScalaVersion] = s match {
+      case ScalaVersionPattern(major, minor) =>
+        for {
+          major <- major.toIntOpt
+          minor <- minor.toIntOpt
+        } yield ScalaVersion(major, minor)
+      case _ ⇒
+        None
+    }
+
+    val scalaVersion = settings
+      .get(ScalaVersionPropertyKey)
+      .toOption
+      .flatMap(parseVersion)
+      .getOrElse(DefaultScalaVersion)
+
+    // log a warning if using the default scala version
+    if (scalaVersion == DefaultScalaVersion)
+      logger.warn(
+        s"[sonar-scala] The '$ScalaVersionPropertyKey' is not properly set or is missing, using the default value: 's$DefaultScalaVersion'"
```

Comment:
`'s$DefaultScalaVersion'` - the `s` at the beginning is redundant.

Reply:
O god my dyslexia again. Sorry for that.